src/main.cpp

```diff
@@ -1505,14 +1483,39 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
         }
     }
 
-    auto valueBalanceOrchard = orchard_bundle.GetValueBalance();
+    // nSpendsSapling, nOutputsSapling, and nActionsOrchard MUST all be less than 2^16
+    size_t max_elements = (1 << 16) - 1;
+    if (tx.vShieldedSpend.size() > max_elements) {
+        return state.DoS(
+            dosLevelPotentiallyRelaxing,
```

Comment:
`dosLevelPotentiallyRelaxing` is only relevant to contextual rules (that can change with block height), so this isn't defined here (and is causing the build to fail):
```suggestion
            100,
```